src/class-user-interface.php

```diff
@@ -0,0 +1,249 @@
+
+	/**
+	 * Admin notices object.
+	 *
+	 * @var Admin_Notices
+	 */
+	private $admin_notices;
+
+	/**
+	 * Initializes the class.
+	 */
+	public function __construct() {
+		global $pagenow;
+		$this->pagenow = $pagenow;
+
+		$this->link_builder                     = new Rewrite_Republish_Link_Builder();
+		$this->rewrite_and_republish_row_action = new Row_Action( $this->link_builder );
+		$this->admin_notices                    = new Admin_Notices();
+
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_scripts' ] );
+		\add_action( 'admin_enqueue_scripts', [ $this, 'should_previously_used_keyword_assessment_run' ], 9 );
+
+		\add_filter( 'post_row_actions', [ $this->rewrite_and_republish_row_action, 'add_action_link' ], 11, 2 );
+		\add_filter( 'page_row_actions', [ $this->rewrite_and_republish_row_action, 'add_action_link' ], 11, 2 );
+		\add_action( 'post_submitbox_start', [ $this, 'add_rewrite_and_republish_post_button' ] );
+		\add_action( 'wp_before_admin_bar_render', [ $this, 'admin_bar_render' ] );
+
+		\add_action( 'admin_init', [ $this, 'add_bulk_filters' ] );
+	}
+
+	/**
+	 * Disables the Yoast SEO PreviouslyUsedKeyword assessment for posts duplicated for Rewrite & Republish.
+	 *
+	 * @return void
+	 */
+	public function should_previously_used_keyword_assessment_run() {
+		if ( ! \in_array( $this->pagenow, [ 'post.php', 'post-new.php' ], true ) ) {
+			return;
+		}
+
+		if ( ! $this->post ) {
+			$this->post = \get_post();
+		}
+
+		$skip_assessment = \get_post_meta( $this->post->ID, '_dp_is_rewrite_republish_copy', true );
+
+		if ( ! empty( $skip_assessment ) ) {
+			\add_filter( 'wpseo_previously_used_keyword_active', '__return_false' );
+		}
+	}
+
+	/**
+	 * Enqueues the necessary JavaScript code for the block editor.
+	 *
+	 * @return void
+	 */
+	public function enqueue_block_editor_scripts() {
+		\wp_enqueue_script(
+			'duplicate_post_edit_script',
+			\plugins_url( \sprintf( 'js/dist/duplicate-post-edit-%s.js', Utils::flatten_version( DUPLICATE_POST_CURRENT_VERSION ) ), DUPLICATE_POST_FILE ),
+			[
+				'wp-blocks',
+				'wp-element',
+				'wp-i18n',
+			],
+			DUPLICATE_POST_CURRENT_VERSION,
+			true
+		);
+
+		\wp_localize_script(
+			'duplicate_post_edit_script',
+			'duplicatePostRewriteRepost',
+			[
+				'permalink' => $this->get_rewrite_republish_permalink(),
+				'rewriting' => ( ! empty( $_REQUEST['rewriting'] ) ) ? 1 : 0,  // phpcs:ignore WordPress.Security.NonceVerification
+			]
+		);
+	}
+
+	/**
+	 * Generates a Rewrite & Republish permalink for the current post.
+	 *
+	 * @return string The permalink. Returns empty if the post hasn't been published yet.
+	 */
+	public function get_rewrite_republish_permalink() {
+		if ( ! $this->post ) {
+			$this->post = \get_post();
+		}
+
+		// phpcs:ignore WordPress.PHP.YodaConditions
```

Comment:
I think this phpcs:ignore can be removed now that Yoda conditions aren't required